machine_learning/neural_network.cpp

```diff
@@ -0,0 +1,719 @@
+                                const std::vector< std::vector <std::valarray <double>>> &Y) {
+                    std::cout << "INFO: Evaluation Started" << std::endl;
+                    double acc = 0, loss = 0; // intialize performance metrics with zero
+                    for(size_t i = 0; i < X.size(); i++) { // For every sample in input
+                        // Get predictions
+                        std::vector<std::valarray<double>> pred = this -> single_predict(X[i]);
+                        // If predicted class is correct 
+                        if(argmax(pred) == argmax(Y[i])) {
+                            acc += 1; // Increment accuracy
+                        }
+                        // Calculating loss - Mean Squared Error
+                        loss += sum(apply_function((Y[i] - pred), 
+                                    neural_network::util_functions::square) * 0.5);
+                    }
+                    acc /= X.size(); // Averaging accuracy
+                    loss /= X.size(); // Averaging loss
+                    // Prinitng performance of the model
+                    std::cout << "Evaluation: Loss: " << loss;
+                    std::cout << ", Accuracy: " << acc << std::endl;
+                    return;
+                }
+
+                /**
+                 * Function to evaluate model on data stored in csv file
+                 * @param file_name csv file name
+                 * @param last_label flag for whether label is in first or last column
+                 * @param normalize flag for whether to normalize data 
+                 */                
+                void evaluate_from_csv (const std::string &file_name, 
+                                     const bool &last_label,
+                                     const bool &normalize) {
+                    // Getting training data from csv file
+                    auto data = this -> get_XY_from_csv(file_name, last_label, normalize);
+                    // Evaluating model
+                    this -> evaluate(data.first, data.second);
+                    return;
+                }                
+
+                /**
+                 * Function to save current model.
+                 * @param file_name file name to save model (*.model)
+                 */
+                void save_model (const std::string &_file_name) {
+                    std::string file_name = _file_name;
+                    // Adding ".model" extension if it is not already there in name
+                    if(file_name.find(".model") == file_name.npos) {
+                        file_name += ".model";
+                    }
+                    std::ofstream out_file; // Ofstream to write in file
+                    // Open file in out|trunc mode
+                    out_file.open(file_name.c_str(), std::ofstream::out | std::ofstream::trunc);
+                    /*
```

Comment:
```suggestion
                    /**
```
Excellent 👍 
Adding the two stars makes this doc block to appear in the function documentation.